### ISSUE TITLE
functionality test before starting the execution

### ISSUE THREAD

**Issue by Harkishen-Singh** [zairza-cetb/bench-routes#145]
Bench-routes provides support for multiple platforms. Features like monitoring can be performed on all platforms. However, those functionalities which require the help of subprocess (say `ping`, `flood ping`) are dependent on the command support offered by the operating system. 

We need to test these supported functionalities and accordingly enable them in the respective operating system. Such tests would be performed at every start and will accordingly inform the user about the in-compatibility of that feature in the UI.

The best example of this would be running bench-routes in windows environment. Functionalities would be disabled accordingly to the availability.

**Comment by Pratik1234agarwal** [comment]
Hey I would like to work on this issue.
Where can I get started?

**Comment by Harkishen-Singh** [comment]
@Pratik1234agarwal we use shell to run ping like commands. If a particular command is not supported, mark that as unusable so that users don't get blocked for that functionality (due to errors).

**Comment by Pratik1234agarwal** [comment]
Okay @Harkishen-Singh  I am working on it. I will update you soon.

**Comment by Pratik1234agarwal** [comment]
Is there any docs there to understand the functionality of how to use each feature.

**Comment by Harkishen-Singh** [comment]
There aren't any at the moment. We will update them sometime later. You can get some ideas related to this if you search in the code for exec.Command().

**Comment by ruddi10** [comment]
@Harkishen-Singh can I work on this?

**Comment by ruddi10** [comment]
> @Pratik1234agarwal we use shell to run ping like commands. If a particular command is not supported, mark that as unusable so that users don't get blocked for that functionality (due to errors).

@Harkishen-Singh just wanted to confirm that incase the ping command isnt available then shouldnt the backend send a blank response instead of throwing errors and to make sure that the error is due to unavailability we can send some string/boolean alongwith it.

**Comment by Harkishen-Singh** [comment]
That can be a way. rather, the tsdb should not get any new blocks so that we never commit those stale chains.

**Comment by ruddi10** [comment]
@Harkishen-Singh can you please explain this a much more it will be helpful

**Comment by Tushar3099** [comment]
@Harkishen-Singh IMHO, I guess we can approach like this : 
- We can make a route, which the frontend will hit before rendering the application.
- In the backend, we can check for the functionality test and send it to the frontend.
- The frontend will show but disable the Tabs whose functionality is not supported.
- This will prevent the user from making any request whose functionality is not supported and prevents any stale chains from committing to the Chainset.

I want to work on this issue.

**Comment by ruddi10** [comment]
@Tushar3099  I am already working on this kindly work on some other issue

**Comment by Harkishen-Singh** [comment]
> ping command isnt available then shouldnt the backend send a blank response

Yes, but it should not send any response (rather warn and stop the ping process, otherwise its just using resources without any reason), If it sends a blank response, we are eating space for trivial data.
